Modules/Core/Common/include/itkImageBase.hxx

```diff
@@ -398,6 +398,38 @@ ImageBase<VImageDimension>::VerifyRequestedRegion()
 }
 
 
+template <unsigned int VImageDimension>
+bool
+ImageBase<VImageDimension>::SamePhysicalSpaceAs(const ImageBase * otherImage,
+                                                double            coordinateTolerance,
+                                                double            directionTolerance) const
+{
+  // check that the image occupy the same physical space, and that
+  // each index is at the same physical location
+
+  // tolerance for origin and spacing depends on the size of pixel
+  // tolerance for directions a fraction of the unit cube.
+  const SpacePrecisionType coordinateTol =
+    itk::Math::abs(coordinateTolerance * this->GetSpacing()[0]); // use first dimension spacing
+
+  return this->GetOrigin().GetVnlVector().is_equal(otherImage->GetOrigin().GetVnlVector(), coordinateTol) &&
+         this->GetSpacing().GetVnlVector().is_equal(otherImage->GetSpacing().GetVnlVector(), coordinateTol) &&
+         this->GetDirection().GetVnlMatrix().as_ref().is_equal(otherImage->GetDirection().GetVnlMatrix().as_ref(),
+                                                               directionTolerance);
+}
+
+
+template <unsigned int VImageDimension>
+bool
+ImageBase<VImageDimension>::SameImageGridAs(const ImageBase * otherImage,
```

Comment:
Please consider declaring the `otherImage` parameter as a reference (`const ImageBase &`).